numba/tests/test_api.py

```diff
@@ -28,5 +29,46 @@ def test_numba_module(self):
         numba.__version__  # not in __all__
 
 
+class TestJitDecorator(TestCase):
+    """
+    Test the jit and njit decorators
+    """
+    def test_jit_nopython_forceobj(self):
+        with self.assertRaises(ValueError):
+            jit(nopython=True, forceobj=True)
+
+        def py_func(x):
+            return(x)
+
+        jit_func = jit(nopython=True)(py_func)
+        jit_func(1)
+        # Check length of nopython_signatures to check
+        # which mode the function was compiled in
+        self.assertEqual(len(jit_func.nopython_signatures), 1)
+
+        jit_func = jit(forceobj=True)(py_func)
+        jit_func(1)
+        self.assertEqual(len(jit_func.nopython_signatures), 0)
+
+    def test_njit_nopython_forceobj(self):
+        with self.assertWarns(RuntimeWarning):
+            njit(forceobj=True)
+
+        with self.assertWarns(RuntimeWarning):
+            njit(nopython=True)
```

Comment:
There's a preference in the Numba tests to be explicit where possible. I think that something along the lines of:
```python
with warnings.catch_warnings(record=True) as w:
  warnings.simplefilter('always', RuntimeWarning)
  <thing to trigger warning>
self.assertEqual(len(w), 1)
self.assertIn(<expected message>, str(w[0].message))
```
could help with checking the expected warning messages are raised?